kong/cache.lua

```diff
@@ -0,0 +1,164 @@
+local kong_mlcache = require "kong.mlcache"
+
+
+local type    = type
+local max     = math.max
+local ngx_log = ngx.log
+local ngx_now = ngx.now
+local ERR     = ngx.ERR
+local NOTICE  = ngx.NOTICE
+local DEBUG   = ngx.DEBUG
+
+
+local SHM_CACHE = "cache"
```

Comment:
Alright, let's take advantage of the fact that 0.11 will have many updates to the Nginx config template already to chose a better name, and group the breaking changes in one.

I suggest we prefix all of our shms with `kong_*`. How about: `kong_db_cache` for this one?